src/blog/_posts/zonal-isolation-v1.md

```diff
@@ -0,0 +1,125 @@
+In the event of an outage, such as a bad deployment, customers may want to drain workers from a specific zone to mitigate the impact. There are two types of drains that can occur:
+1. **Implicit drain:** Workers from a zone completely stop operating, either due to failures or manual shutdowns.
+2. **Explicit drain:** Customers explicitly mark workers from a zone as drained.
+
+Within cadence-matching, a list of pollers is maintained for each tasklist, tracking the zone from which the pollers originate. An implicit drain can be detected by checking whether there are any active pollers from a particular zone. If a zone is drained (either explicitly or implicitly), workflows that were started in that zone will be reassigned and spread to workers in the remaining healthy zones.
+##### How to drain a zone explicitly?
+Explicitly draining a zone can be done at two levels:
+1. **Domain-level drain:** This applies to a specific domain, allowing the zone to be drained only for workflows within that domain.
+2. **Cluster-level drain:** This applies to the entire Cadence cluster, draining the zone for all domains and workflows within the cluster.
+
+For domain-level drain, the draining status is stored in [`domains`](https://github.com/uber/cadence/blob/v1.2.13/schema/cassandra/cadence/schema.cql#L412) table. For cluster-level drain, the status is stored in [`cluster_config`](https://github.com/uber/cadence/blob/v1.2.13/schema/cassandra/cadence/schema.cql#L498) table.
+
+## How to enable Zonal Isolation?
+### Server Update
+In order to enable this feature, please upgrade Cadence server to [v1.2.1](https://github.com/uber/cadence/releases/tag/v1.2.1) or later.
+
+**NOTE:** If you're not using the provided main binary located in `cmd/server`, you must adopt this [middleware](https://github.com/uber/cadence/blob/v1.2.1/common/rpc/middleware.go#L188). Depends on the appoach to [determine the origin zone of requests](#determine-the-zone-of-a-workflow-and-workers), you can adopt this [middleware](https://github.com/uber/cadence/blob/v1.2.1/common/rpc/middleware.go#L229) or build your own middleware using [`partition.ContextWithConfig`](https://github.com/uber/cadence/blob/v1.2.1/common/partition/context.go#L42) function injecting origin zone into the context.
+
+This feature is controled by 2 dynamic config properties.
+- `system.allIsolationGroups`: This property provides the list of available zones within a region.
+- `system.enableTasklistIsolation`: This property enables Zonal Isolation at domain level.
+It’s important to note that any update to the `system.allIsolationGroups` property requires a restart of the cadence-matching service for the changes to take effect. This ensures that the new zone configuration is properly loaded and applied. However, the `system.enableTasklistIsolation` property can be updated dynamically without requiring a restart, making it more flexible for enabling or disabling Zonal Isolation on a per-domain basis.
```

Comment:
> requiring a restart

@davidporter-id-au After your changes for auto discovery, will it require restarting matching to recognize changes to `system.allIsolationGroups` ?

Reply:
no